exec-analysis/src/main/java/com/capitalone/dashboard/exec/collector/PipelineCollector.java

```diff
@@ -0,0 +1,126 @@
+        return MetricCollectionStrategy.AVERAGE;
+    }
+
+    @Override
+    protected CollectorItemMetricDetail getCollectorItemMetricDetail(List<Row> rowList, MetricType metricType) {
+        CollectorItemMetricDetail collectorItemMetricDetail = new CollectorItemMetricDetail();
+        if (CollectionUtils.isEmpty(rowList)) {
+            return collectorItemMetricDetail;
+        }
+        collectorItemMetricDetail.setType(metricType);
+        for (Row row : rowList) {
+            updateCollectorItemMetricDetail(collectorItemMetricDetail, row);
+        }
+        return collectorItemMetricDetail;
+    }
+
+    private void updateCollectorItemMetricDetail(CollectorItemMetricDetail collectorItemMetricDetail, Row itemRow) {
+        Date timeWindowDt = itemRow.getAs("timeWindow");
+        LOGGER.info("TimeWindow:" +timeWindowDt );
+        LOGGER.info("itemRow :" + itemRow);
+        Collection<Object> javaCollection = JavaConversions.asJavaCollection(((WrappedArray) itemRow.getAs("prodStageList")).toList());
+
+        Optional.ofNullable(javaCollection)
+                .orElseGet(Collections::emptyList).stream().map(m -> (GenericRowWithSchema) m).forEach(genericRowWithSchema -> {
+            Long pipelineTimeL = genericRowWithSchema.getAs("timestamp");
+            Date dateObj = new Timestamp(new Date(pipelineTimeL).getTime());
+            LOGGER.info("Date Object :" + dateObj);
+            Long scmTimeL = genericRowWithSchema.getAs("scmCommitTimestamp");
+            Long pipelineTimeAfterIgnore = pipelineTimeL/1000;
+            Long scmTimeAfterIgnore = scmTimeL/1000;
+            try {
+                Long diffTimestamp = Math.abs(pipelineTimeAfterIgnore - scmTimeAfterIgnore);
+                String strTimestampInsec = Long.toString(diffTimestamp);
+                double value = Double.parseDouble(strTimestampInsec);
+                MetricCount mc = getMetricCount("", value, "pipeline-lead-time");
+                if (mc != null) {
+                    collectorItemMetricDetail.setStrategy(getCollectionStrategy());
+                    collectorItemMetricDetail.addCollectorItemMetricCount(dateObj, mc);
+                    collectorItemMetricDetail.setLastScanDate(dateObj);
+                }
+            } catch (NumberFormatException e) {
+                LOGGER.info("Exception: Not a number, 'value' = " + scmTimeAfterIgnore, e);
+            }
+        });
+    }
+
+    @Override
+    protected MetricCount getMetricCount(String level, double value, String name) {
+        MetricCount mc = new MetricCount();
+        Map<String, String> label = new HashMap<>();
+        String metricLabel = getMetricLabel(name);
+        if(metricLabel != null) {
+            label.put("type", metricLabel);
+            mc.setLabel(label);
+            mc.setValue(value);
+        }
+        return mc;
+    }
+
+    private String getMetricLabel(String inputMetricLabel) {
+        if(inputMetricLabel.equals("pipeline-lead-time")) {
```

Comment:
please ensure the `inputMetricLabel` is null safe or consider using StringUtils.equals()

Reply:
Usually, I took an example from StaticCode analysis, but as suggested updated code with StringUtils.equal()